tests/pubsub/check_pubsub_publish_mqtt.c

```diff
@@ -0,0 +1,174 @@
+                                               (UA_UadpNetworkMessageContentMask)UA_UADPNETWORKMESSAGECONTENTMASK_PAYLOADHEADER);
+        writerGroupConfig.messageSettings.content.decoded.data = writerGroupMessage;
+
+        /* configure the mqtt publish topic */
+        UA_BrokerWriterGroupTransportDataType brokerTransportSettings;
+        memset(&brokerTransportSettings, 0, sizeof(UA_BrokerWriterGroupTransportDataType));
+        brokerTransportSettings.queueName = UA_STRING(PUBLISHER_TOPIC);
+        brokerTransportSettings.resourceUri = UA_STRING_NULL;
+        brokerTransportSettings.authenticationProfileUri = UA_STRING_NULL;
+
+        brokerTransportSettings.requestedDeliveryGuarantee = UA_BROKERTRANSPORTQUALITYOFSERVICE_BESTEFFORT;
+
+        UA_ExtensionObject transportSettings;
+        memset(&transportSettings, 0, sizeof(UA_ExtensionObject));
+        transportSettings.encoding = UA_EXTENSIONOBJECT_DECODED;
+        transportSettings.content.decoded.type = &UA_TYPES[UA_TYPES_BROKERWRITERGROUPTRANSPORTDATATYPE];
+        transportSettings.content.decoded.data = &brokerTransportSettings;
+
+        writerGroupConfig.transportSettings = transportSettings;
+        retval = UA_Server_addWriterGroup(server, connectionIdent, &writerGroupConfig, &writerGroupIdent);
+        ck_assert_int_eq(retval, UA_STATUSCODE_GOOD);
+        retval = UA_Server_setWriterGroupOperational(server, writerGroupIdent);
+        ck_assert_int_eq(retval, UA_STATUSCODE_GOOD);
+
+        // add DataSetWriter
+        UA_NodeId dataSetWriterIdent;
+        UA_DataSetWriterConfig dataSetWriterConfig;
+        memset(&dataSetWriterConfig, 0, sizeof(dataSetWriterConfig));
+        dataSetWriterConfig.name = UA_STRING("DataSetWriter 1");
+        dataSetWriterConfig.dataSetWriterId = 62541;
+        dataSetWriterConfig.keyFrameCount = 10;
+        dataSetWriterConfig.transportSettings = transportSettings;
+
+        retval = UA_Server_addDataSetWriter(server, writerGroupIdent, publishedDataSetIdent,
+                                            &dataSetWriterConfig, &dataSetWriterIdent);
+        ck_assert_int_eq(retval, UA_STATUSCODE_GOOD);
+
+        UA_WriterGroup *wg = UA_WriterGroup_findWGbyId(server, writerGroupIdent);
+        ck_assert(wg != 0);
+        UA_WriterGroup_publishCallback(server, wg);
+    } END_TEST
+
+int main(void) {
+    TCase *tc_pubsub_publish_mqtt = tcase_create("PubSub publish mqtt");
+    tcase_add_checked_fixture(tc_pubsub_publish_mqtt, setup, teardown);
+    tcase_add_test(tc_pubsub_publish_mqtt, SinglePublishDataSetField);
+
+    Suite *s = suite_create("PubSub publishing via mqtt");
+    suite_add_tcase(s, tc_pubsub_publish_mqtt);
+
+    SRunner *sr = srunner_create(s);
+    srunner_set_fork_status(sr, CK_NOFORK);
+    srunner_run_all(sr,CK_NORMAL);
+    int number_failed = srunner_ntests_failed(sr);
+    srunner_free(sr);
+    return (number_failed == 0) ? EXIT_SUCCESS : EXIT_FAILURE;
+}
```

Comment:
Is there an aspect which is not covered by the check_pubsub_subscribe test? Maybe we should combine it to one unit test.